lib/cerner/oauth1a/access_token_agent.rb

```diff
@@ -143,6 +146,26 @@ def retrieve(principal: nil, ignore_cache: false)
         access_token
       end
 
+      # Public: Constructs an AccessToken using the realm of this agent and the value of an HTTP
+      # Authorization Header based on the OAuth HTTP Authorization Scheme (https://oauth.net/core/1.0a/#auth_header).
+      #
+      # value - A String containing the HTTP Authorization Header value.
+      #
+      # Returns an AccessToken.
+      #
+      # Raises a Cerner::OAuth1a::OAuthError with a populated oauth_problem and realm of this agent
+      # if any of the parameters in the value are invalid.
+      def from_authorization_header(value)
```

Comment:
Is there a better name for this or a better way to provide realm validation against an agent when parsing an access token from the authorization header?

Reply:
Why wouldn't all of this be in the AccessToken.from_authorization_header? The current design is meant to be that parsing the header gives you either a valid AccessToken or an OAuthError. I'd like to see all of this just go into that method.

Reply:
Without access to the agent there isn't a way to directly validate the realm value..

`AccessToken.from_authorization_header` could be updated to validate against the optional `realm` parameter if present, and include documentation that suggests providers set the `realm` parameter from the agent they want to validate against. This would still provide validation when used correctly.

Reply:
Can we move this into the AccessToken#authenticate method? That's really what I'd like to have fail. 
AccessToken#from_authorization_header is a parsing method and should only fail on parsing errors. AccessToken#authenticate is a processing method and should fail when the token isn't valid.

Why didn't you add realm to the AccessToken? Isn't that considered part of the token?